src/main/java/com/spotify/confidence/EventSenderEngineImpl.java

```diff
@@ -44,13 +52,34 @@ public void run() {
         }
         final var numberOfPendingEvents = eventStorage.write(event);
         if (numberOfPendingEvents >= maxBatchSize) {
-          eventStorage.createBatch();
-          uploadQueue.add(UPLOAD_SIG);
+          flush();
+        } else {
+          scheduleFlush();
         }
       }
     }
   }
 
+  private void scheduleFlush() {
+    // Cancel the existing scheduled task if it exists
+    if (pendingFlush != null && !pendingFlush.isDone()) {
+      pendingFlush.cancel(false);
```

Comment:
Don't think these `cancel(false)` will do anything:
![Screenshot 2024-04-08 at 11 40 43](https://github.com/spotify/confidence-openfeature-provider-java/assets/12396573/cf2ae743-1458-47a9-ba87-13331c68e4aa)

But if you are making the flush thread safe I don't think it's should be needed?